lib/puppet/resource.rb

```diff
@@ -11,9 +11,12 @@ class Puppet::Resource
   include Puppet::Util::Tagging
 
   include Enumerable
-  attr_accessor :file, :line, :catalog, :exported, :virtual, :validate_parameters, :strict
+  attr_accessor :file, :line, :catalog, :exported, :virtual, :strict
   attr_reader :type, :title
 
+  # @deprecated not used
```

Comment:
You cannot know that - it may be unused in the puppet code base but it may be used elsewhere.